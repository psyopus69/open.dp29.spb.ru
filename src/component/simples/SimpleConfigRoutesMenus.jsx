import React from "react";
import {PropTypes} from "prop-types";

const SimpleConfigRoutesMenus = (props) => {
  const {renderRoutes, renderCustomRoute, renderLink, renderCustomLink} = props;
  const Custom = () => {
    return <div>@@@@ CustomComponent @@@@</div>;
  };
  return <React.Fragment>
    <div>MenusCustom:</div>{renderCustomLink({
      to: `/custom`,
      name: `custom`,
    })}
    <div>Menus /about/secret:</div>{renderLink(`/about/secret`)}
    <div>MenusAll:</div>{renderLink()}
    <div>RoutesCustom:</div>{renderCustomRoute({
      path: `/custom`,
      component: Custom,
    })}
    <div>RoutesAll:</div>{renderRoutes()}
  </React.Fragment>;
};
SimpleConfigRoutesMenus.propTypes = {
  renderRoutes: PropTypes.func.isRequired,
  renderCustomRoute: PropTypes.func.isRequired,
  renderLink: PropTypes.func.isRequired,
  renderCustomLink: PropTypes.func.isRequired,
};
export {
  SimpleConfigRoutesMenus,
};
