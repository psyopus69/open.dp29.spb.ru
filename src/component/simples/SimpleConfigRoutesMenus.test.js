import renderer from "react-test-renderer";
import React from "react";
import {SimpleConfigRoutesMenus} from "./SimpleConfigRoutesMenus";

it(`all methods of 2 HOCK's withRoutes & withMenu`, () => {
  const tree = renderer.create(<SimpleConfigRoutesMenus renderRoutes={jest.fn()} renderCustomRoute={jest.fn()} renderLink={jest.fn()} renderCustomLink={jest.fn()}/>);
  expect(tree).toMatchSnapshot();
});
