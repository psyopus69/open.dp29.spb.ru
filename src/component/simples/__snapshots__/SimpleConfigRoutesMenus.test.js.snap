// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all methods of 2 HOCK's withRoutes & withMenu 1`] = `
Array [
  <div>
    MenusCustom:
  </div>,
  <div>
    Menus /about/secret:
  </div>,
  <div>
    MenusAll:
  </div>,
  <div>
    RoutesCustom:
  </div>,
  <div>
    RoutesAll:
  </div>,
]
`;
